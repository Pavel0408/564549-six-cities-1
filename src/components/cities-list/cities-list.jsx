import PropTypes from "prop-types";
import React, {PureComponent} from "react";
import {City} from "../city/city";

export class CitiesList extends PureComponent {
  render() {
    const {cities, cityClickHandler, cityName} = this.props;
    return <div className="cities tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.map((cityItem, i) => {
            return <City
              key={`city-${i}`}
              city={cityItem}
              isActive={cityItem === cityName}
              cityClickHandler={cityClickHandler}
            />;
          })
          }
        </ul>
      </section>
    </div>;
  }
}

CitiesList.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string),
  cityClickHandler: PropTypes.func,
  cityName: PropTypes.string
};

