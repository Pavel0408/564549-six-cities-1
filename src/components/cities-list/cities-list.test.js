import React from 'react';
import renderer from 'react-test-renderer';
import {CitiesList} from './cities-list';

const activeCity = `Amsterdam`;
const cityClickHandler = () => {
};
const cities = [
  `Paris`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

describe(`testing CitiesList render`, () => {
  it(`CitiesList is render correctly`, () => {
    const tree = renderer.create(<CitiesList
      cities={cities}
      activeCity={activeCity}
      cityClickHandler={cityClickHandler}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
