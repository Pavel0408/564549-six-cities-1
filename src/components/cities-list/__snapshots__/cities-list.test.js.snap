// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CitiesList render CitiesList is render correctly 1`] = `
<div
  className="cities tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item"
          onClick={[Function]}
        >
          <span>
            Paris
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item"
          onClick={[Function]}
        >
          <span>
            Brussels
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item"
          onClick={[Function]}
        >
          <span>
            Hamburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item"
          onClick={[Function]}
        >
          <span>
            Dusseldorf
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
